pkg/executor/join/hash_join_stats.go

```diff
@@ -0,0 +1,261 @@
+		buf.WriteString(", fetch:")
+		buf.WriteString(execdetails.FormatDuration(e.fetchAndBuildHashTable - e.hashStat.buildTableElapse))
+		buf.WriteString(", build:")
+		buf.WriteString(execdetails.FormatDuration(e.hashStat.buildTableElapse))
+		buf.WriteString("}")
+	}
+	if e.probe > 0 {
+		buf.WriteString(", probe:{concurrency:")
+		buf.WriteString(strconv.Itoa(e.concurrent))
+		buf.WriteString(", total:")
+		buf.WriteString(execdetails.FormatDuration(time.Duration(e.fetchAndProbe)))
+		buf.WriteString(", max:")
+		buf.WriteString(execdetails.FormatDuration(time.Duration(atomic.LoadInt64(&e.maxFetchAndProbe))))
+		buf.WriteString(", probe:")
+		buf.WriteString(execdetails.FormatDuration(time.Duration(e.probe)))
+		// fetch time is the time wait fetch result from its child executor,
+		// wait time is the time wait its parent executor to fetch the joined result
+		buf.WriteString(", fetch and wait:")
+		buf.WriteString(execdetails.FormatDuration(time.Duration(e.fetchAndProbe - e.probe)))
+		if e.hashStat.probeCollision > 0 {
+			buf.WriteString(", probe_collision:")
+			buf.WriteString(strconv.FormatInt(e.hashStat.probeCollision, 10))
+		}
+		buf.WriteString("}")
+	}
+	return buf.String()
+}
+
+func (e *hashJoinRuntimeStats) Clone() execdetails.RuntimeStats {
+	return &hashJoinRuntimeStats{
+		fetchAndBuildHashTable: e.fetchAndBuildHashTable,
+		hashStat:               e.hashStat,
+		fetchAndProbe:          e.fetchAndProbe,
+		probe:                  e.probe,
+		concurrent:             e.concurrent,
+		maxFetchAndProbe:       e.maxFetchAndProbe,
+	}
+}
+
+func (e *hashJoinRuntimeStats) Merge(rs execdetails.RuntimeStats) {
+	tmp, ok := rs.(*hashJoinRuntimeStats)
+	if !ok {
+		return
+	}
+	e.fetchAndBuildHashTable += tmp.fetchAndBuildHashTable
+	e.hashStat.buildTableElapse += tmp.hashStat.buildTableElapse
+	e.hashStat.probeCollision += tmp.hashStat.probeCollision
+	e.fetchAndProbe += tmp.fetchAndProbe
+	e.probe += tmp.probe
+	if e.maxFetchAndProbe < tmp.maxFetchAndProbe {
+		e.maxFetchAndProbe = tmp.maxFetchAndProbe
+	}
+}
+
+type hashStatistic struct {
+	// NOTE: probeCollision may be accessed from multiple goroutines concurrently.
+	probeCollision   int64
+	buildTableElapse time.Duration
+}
+
+func (s *hashStatistic) String() string {
+	return fmt.Sprintf("probe_collision:%v, build:%v", s.probeCollision, execdetails.FormatDuration(s.buildTableElapse))
+}
+
+type hashJoinRuntimeStatsV2 struct {
+	concurrent     int
+	probeCollision int64
+
+	fetchAndBuildHashTable int64
+
+	partitionData  int64
+	buildHashTable int64
+	probe          int64
+	fetchAndProbe  int64
+
+	maxPartitionData  int64
+	maxBuildHashTable int64
+	maxProbe          int64
+	maxFetchAndProbe  int64
+
+	maxPartitionDataForCurrentRound  int64
+	maxBuildHashTableForCurrentRound int64
+	maxProbeForCurrentRound          int64
+	maxFetchAndProbeForCurrentRound  int64
+}
+
+func setMaxValue(addr *int64, currentValue int64) {
```

Comment:
I think `setMaxFetchAndProbeTime` can be replaced by this function?

Reply:
> I think `setMaxFetchAndProbeTime` can be replaced by this function?

done